src/cryptsetup/cryptsetup-generator.c

```diff
@@ -152,7 +152,9 @@ static int create_disk(
                               f);
         } else
                 fprintf(f,
-                        "RequiresMountsFor=%s\n",
+                        "RequiresMountsFor=%s\n"
+                        "Requires=systemd-tmpfiles-setup-dev.service\n"
```

Comment:
hmm,  we don't need the Requires= here, do we? I mean the unit is pulled in anyway, this is merely an ordering issue.

Generally we try to minimize coupling, i.e. be as loose as possible and as strict as necessary. The Requires= dep appears to be unnecessary to me, the After= should suffice, no?

Reply:
It needs both, actually. It cannot work without Requires= because otherwise we don't have /dev/loop-control around.

Reply:
I can test it and find out the definitive answer tomorrow when I get back to the machine exhibiting this issue :)

Reply:
No need for that; After= is enough. The service in question is always pulled in because it is part of low-level boot. The question is whether it should still be listed for completeness sake. I am personally on the yes side of that. A service file should always list its dependencies it really has and it would be systemd's job to ignore unnecessary ones. But also, not my call to make :D

Reply:
@medhefgo well, but whether that service is really a dependency is really debatable. I mean, i use cryptsetup, but not for loopback stuff, hence for me it really isn't. also, if creating some device node fails, this should not matter for cryptsetup usually, hence Requires= is wrong anyway...

The way to look at it I think is always to consider a user who booted into emergency mode and now piecemeal starts services, and only the one he really needs. If he decides to start cryptsetup, does that really mean he should also get the wants stuff? I am not convinced...

Reply:
This is for the code path for loopback devices only. /dev crypto devices don't get this dependency.

Reply:
What happens if `systemd-tmpfiles-setup-dev.service` is masked and `modules_load=loop` is set to kernel command line as you suggested in #9308?

Reply:
@yuwata If people wanna shoot themselves in the foot ^^^^ customize their system, let them. It's not our job to make things easier for them.

Anyway, you did make me realize that our dependency really is /dev/loop-control. So Requires+After=dev-loop\x2dcontrol.device *should* do the trick much more correctly. @lorenzhs  Can you give this dependency a spin on your system?

Reply:
Not sure if that's a Debian thing but `dev-loop\x2dcontrol.device` isn't a thing on my system:
```sh
$ systemctl status dev-loop\\x2dcontrol.device
● dev-loop\x2dcontrol.device
   Loaded: loaded
   Active: inactive (dead)
```

Reply:
(it also doesn't help to put that in the unit file)

Reply:
Then it probably needs this udev rule too:
KERNEL=="loop-control", TAG+="systemd"

Reply:
> This is for the code path for loopback devices only. /dev crypto devices don't get this dependency.

That is a very good point.

Reply:
> Anyway, you did make me realize that our dependency really is /dev/loop-control. So Requires+After=dev-loop\x2dcontrol.device should do the trick much more correctly. @lorenzhs Can you give this dependency a spin on your system?

Hmm, we don't tag that device with "systemd" in the udev database, which means it never shows up as ".device" unit, and I am not convinced we should change that. So far we put the tag mostly on "real" devices, not API entrypoint stuff. i.e. /dev/null and /dev/random area also not tagged, for the same reason.

hence I think the patch as it is right now actually looks good to me.